### PR TITLE
kvserver: add metrics for raft (re-)proposals

### LINKED ISSUE #105199: kvserver: add metrics for reproposals
We should have metric counters for re-submitting commands to raft:

https://github.com/cockroachdb/cockroach/blob/00b59676e40326ee27c0cb0cbaf50faabb47430a/pkg/kv/kvserver/replica_raft.go#L1490-L1491

and for reproposing under a new LAI:

https://github.com/cockroachdb/cockroach/blob/12888971a1f82f879f8b36dad8ba347550053ffe/pkg/kv/kvserver/replica_application_result.go#L238

Jira issue: CRDB-28921

### PR DESCRIPTION
**Pull request by pav-kv**
This commit adds new metrics:

- `raft.commands.proposed`: commands proposed to Raft by leaseholders
- `raft.commands.reproposed.unchanged`: commads retried/reproposed to Raft because they take too long to apply (so they might be dropped)
- `raft.commands.reproposed.new-lai`: commands retried/reproposed to Raft because they were committed to Raft out of order (failed the LAI or closed timestamp check)

The `proposed` metric includes both `reproposed` metrics.

Resolves #105199
Epic: none
Release note (ops change): added metrics for raft proposals and reproposals

### CODE REVIEW

File: pkg/kv/kvserver/metrics.go [hunk 1]
```diff
@@ -1221,6 +1221,37 @@ histogram.
...
+likely dropped. Such re-proposals make sure that all proposals are eventually
+committed to Raft and applied.`,
+		Measurement: "Commands",
+		Unit:        metric.Unit_COUNT,
+	}
+	metaRaftCommandsReproposedLAI = metric.Metadata{
+		Name: "raft.commands.reproposed.lai",
+		Help: `Count of Raft commands re-proposed with a later LAI.
+
+This measurement is taken on the leaseholders, and counts retries of proposals
```
Comment (aliher1911) [inline pkg/kv/kvserver/metrics.go:1249]: Maybe instead say what lai change mean for proposal? Like timeout in reproposed.unchanged.

File: pkg/kv/kvserver/metrics.go [hunk 2]
```diff
@@ -1221,6 +1221,37 @@ histogram.
...
+
+This measurement is taken on the leaseholders, and includes both the original
+proposals and re-proposals. This approximates the number of commands submitted
+through Raft.`,
+		Measurement: "Commands",
+		Unit:        metric.Unit_COUNT,
+	}
+	metaRaftCommandsReproposed = metric.Metadata{
+		Name: "raft.commands.reproposed.same",
+		Help: `Count of Raft commands re-proposed without modification.
```
Comment (aliher1911) [inline pkg/kv/kvserver/metrics.go:1236]: I'm not sure metric help is the right place to explain that proposals are done on leaseholder and why we make reproposals. Something like:

Number of Raft commands re-proposed without modification by leaseholder. These re-proposals happen when command is not committed/applied within timeout and is assumed lost and needs to be resent.

**Review by aliher1911 — APPROVED** [review]
Some nits for names/help for metrics.

**Comment by pavelkalinnikov** [comment]
bors r=aliher1911

**Pull request merged**